pkgs/servers/ldap/389/default.nix

```diff
@@ -1,32 +1,108 @@
-{ lib, stdenv, fetchurl, fetchpatch, autoreconfHook, pkg-config, doxygen, perl, pam, nspr, nss, openldap
-, db, cyrus_sasl, svrcore, icu, net-snmp, libkrb5, pcre, perlPackages, libevent, openssl, python3
+{ stdenv
+, autoreconfHook
+, fetchFromGitHub
+, lib
+
+, bzip2
+, cmocka
+, cracklib
+, cyrus_sasl
+, db
+, doxygen
+, icu
+, libevent
+, libkrb5
+, lm_sensors
+, net-snmp
+, nspr
+, nss
+, openldap
+, openssl
+, pcre
+, perl
+, perlPackages
+, pkg-config
+, python3
+, svrcore
+, zlib
+
+, enablePamPassthru ? true
+, pam
+
+, enableCockpit ? true
+, rsync
+
+, enableDna ? true
+, enableLdapi ? true
+, enableAutobind ? false
+, enableAutoDnSuffix ? false
+, enableBitwise ? true
+, enableAcctPolicy ? true
+, enablePosixWinsync ? true
```

Comment:
Do we really need all those options or could we drop some of them?

Reply:
I don't really use this, I just wanted to update it since the current version we have in nixpkgs in insecure.

I'd rather we expose these things so it's easy for users to override than not :)

Reply:
If no one uses these switches or tests them in combination they are likely broken. Not a fan of that.

Reply:
I tested toggling them in a number of combinations and they all built fine.

I followed the enable / disable pattern from the fedora pkg.